py/kubeflow/kubeflow/ci/workflow_utils.py

```diff
@@ -0,0 +1,249 @@
+                 test_target_name=None,
+                 **kwargs):
+        self.name = name
+        self.namespace = namespace
+        self.bucket = bucket
+        self.template_label = "argo_test"
+        self.test_target_name = test_target_name
+        self.mkdir_task_name = "make-artifacts-dir"
+
+        # *********************************************************************
+        #
+        # Define directory locations
+        #
+        # *********************************************************************
+
+        # mount_path is the directory where the volume to store the test data
+        # should be mounted.
+        self.mount_path = "/mnt/" + "test-data-volume"
+        # test_dir is the root directory for all data for a particular test
+        # run.
+        self.test_dir = self.mount_path + "/" + self.name
+        # output_dir is the directory to sync to GCS to contain the output for
+        # this job.
+        self.output_dir = self.test_dir + "/output"
+
+        self.artifacts_dir = "%s/artifacts/junit_%s" % (self.output_dir, name)
+
+        # source directory where all repos should be checked out
+        self.src_root_dir = "%s/src" % self.test_dir
+        # The directory containing the kubeflow/kubeflow repo
+        self.src_dir = "%s/kubeflow/kubeflow" % self.src_root_dir
+
+        # Root of testing repo.
+        self.testing_src_dir = os.path.join(self.src_root_dir,
+                                            "kubeflow/testing")
+
+        # Top level directories for python code
+        self.kubeflow_py = self.src_dir
+
+        # The directory within the kubeflow_testing submodule containing
+        # py scripts to use.
+        self.kubeflow_testing_py = "%s/kubeflow/testing/py" % self.src_root_dir
+
+        self.go_path = self.test_dir
+
+    def _build_workflow(self):
+        """Create a scaffolding CR for the Argo workflow"""
+        volumes = [{
+            "name": DATA_VOLUME,
+            "persistentVolumeClaim": {
+                "claimName": NFS_VOLUME_CLAIM
+            },
+        }]
+        if LOCAL_TESTING == "False":
+            volumes.append(CREDENTIALS_VOLUME)
+
+        workflow = {
+            "apiVersion": "argoproj.io/v1alpha1",
+            "kind": "Workflow",
+            "metadata": {
+                "name": self.name,
+                "namespace": self.namespace,
+                "labels": argo_build_util.add_dicts([
+                    {
+                        "workflow": self.name,
+                        "workflow_template": self.template_label,
+                    },
+                    argo_build_util.get_prow_labels()
+                ]),
+            },
+            "spec": {
+                "entrypoint": E2E_DAG_NAME,
+                # Have argo garbage collect old workflows otherwise we overload
+                # the API server.
+                "ttlSecondsAfterFinished": 7 * 24 * 60 * 60,
```

Comment:
In new argo cluster, we installed v2.12.3 argo workflow, this parameter has been deprecated. 

https://github.com/argoproj/argo/blob/master/examples/gc-ttl.yaml

You can instead use ttlStrategy

Reply:
Or make it simpler, you don't need to set it up, we have already configured default workflow spec 

```
configmap -o yaml
apiVersion: v1
data:
  workflowDefaults: |
    metadata:
      annotations:
        argo: workflows
    spec:
      ttlStrategy:
        secondsAfterSuccess: 604800
        secondsAfterCompletion: 604800
        secondsAfterFailure: 604800
      activeDeadlineSeconds: 86400
...
```

Reply:
OK, I'll just omit this field entirely. There no issue by completely omitting the field and the configmap's value will be used right?

Reply:
Yeah default workflow should handle it.